pkg/controllers/provisioning/provisioner.go

```diff
@@ -136,3 +140,22 @@ func (p *Provisioner) Batch(ctx context.Context) (pods []*v1.Pod) {
 		}
 	}
 }
+
+// RemoveScheduled removes pods that have been assigned a node.
+// This check is needed to prevent duplicate binds when a pod is scheduled to a node
+// between the time it was ingested into the scheduler and the time it is included
+// in a provisioner batch.
+func (p *Provisioner) RemoveScheduled(ctx context.Context, pods []*v1.Pod) []*v1.Pod {
```

Comment:
The hIgHeR ORdeR fUNcTioNS naming for this would be FilterProvisionable(ctx, pods)